src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -1608,12 +1608,18 @@ impl Step for Assemble {
         };
 
         if let Some(enzyme_install) = enzyme_install {
-            let src_lib = enzyme_install.join("build/Enzyme/LLVMEnzyme-17.so");
+            let (lib_prefix, lib_ext) = match env::consts::OS {
+                "macos" => ("lib", "dylib"),
+                "windows" => ("", "dll"),
+                _ => ("", "so")
```

Comment:
for linux it would also be lib as prefix, as with the dst_lib.
Also, I assume you will also need to update dst_lib with the prefix?